pkg/api/util/util.go

```diff
@@ -22,6 +22,12 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+const (
```

Comment:
This can be improved a bit more. The idiomatic way to make an enum-like type in go should do the following:
1. Define a type alias: `type source int`
2. Change this list of constants to use that type:
```
const (
  uninitialized source = iota
  setAuthToken
  createAndSetAuthToken
)
```
3. Finally, use that enum type instead of `int`, so declare `initSource source`.